src/implementations/react/src/elements/components/FormElements/Dropdown.js

```diff
@@ -0,0 +1,168 @@
+import React, { Component } from 'react';
+import DropdownAdapter
+  from '../../../adapters/FormElements/DropdownAdapter.js';
+import * as PropTypes from 'prop-types';
+
+const OptionAdapter = DropdownAdapter.Option;
+
+class Option extends Component {
+  static propTypes = {
+    onClick: PropTypes.func.isRequired,
+    value: PropTypes.string.isRequired
+  }
+
+  handleClick = (event) => {
+    this.props.onClick(this.props.value);
+  }
+
+  render() {
+    return (
+      <OptionAdapter {...this.props} onClick={this.handleClick} />
+    )
+  }
+}
+
+class Dropdown extends Component {
+
+  static propTypes = {
+    label: PropTypes.string,
+    instructions: PropTypes.string,
+    placeholder: PropTypes.string,
+    disabled: PropTypes.bool,
+    required: PropTypes.string,
+    value: PropTypes.string,
+    defaultValue: PropTypes.string,
+    options: PropTypes.arrayOf(PropTypes.shape({
+      label: PropTypes.string,
+      value: PropTypes.string
+    })),
+    onChange: PropTypes.func,
+    onBlur: PropTypes.func,
+    onFocus: PropTypes.func,
+    onKeypress: PropTypes.func
+  }
+
+  static defaultProps = {
+    options: [],
+    onChange: () => {}
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      open: false,
+      value: this.getDefaultValue()
+    };
+  }
+
+  getDefaultValue() {
+    const { defaultValue, value } = this.props;
+
+    if (value !== undefined) {
+      return value;
+    } else if (defaultValue !== undefined) {
+      return defaultValue;
+    } else {
+      return '';
+    }
+  }
+
+  getRenderedValue() {
+    const { value } = this.props;
+
+    if (value !== undefined) {
+      return value;
+    } else {
+      return this.state.value;
+    }
+  }
+
+  setSelectedValue = (selectedOptionValue) => {
+    this.props.onChange(selectedOptionValue);
+
+    this.setState({
+      open: false,
+      value: selectedOptionValue
+    });
+  };
+
+  openDropdown = () => {
+    this.setState({ open: true });
+  };
+
+  closeDropdown = () => {
+    this.setState({ open: false });
+  };
+
+  render() {
+    const selectedOption = this.props.options.find(o => o.value === this.getRenderedValue()) || {};
+
+    return (
+      <DropdownAdapter
+        {...this.props}
```

Comment:
I pass props through using the splat operator that aren't directly relevant to this wrapper. Just less repetitive typing, and less to update if those irrelevant props end up changing later.